crates/sdk/src/queries/vp/pos.rs

```diff
@@ -249,6 +254,45 @@ where
     read_all_validator_addresses(ctx.state, epoch)
 }
 
+/// Get liveness information for all consensus validators in the current epoch.
+fn liveness_info<D, H, V, T>(
+    ctx: RequestCtx<'_, D, H, V, T>,
+) -> namada_storage::Result<LivenessInfo>
+where
+    D: 'static + DB + for<'iter> DBIter<'iter> + Sync,
+    H: 'static + StorageHasher + Sync,
+{
+    let epoch = ctx.state.in_mem().last_epoch;
+    let consensus_validators =
+        read_consensus_validator_set_addresses(ctx.state, epoch)?;
+    let params = read_pos_params::<_, governance::Store<_>>(ctx.state)?;
+
+    let mut result = vec![];
+    for validator in consensus_validators {
+        if let Some(pubkey) = get_consensus_key::<_, governance::Store<_>>(
+            ctx.state, &validator, epoch,
+        )? {
+            let comet_address = tm_consensus_key_raw_hash(&pubkey);
+            let sum_liveness_handle = liveness_sum_missed_votes_handle();
+            if let Some(missed_votes) =
+                sum_liveness_handle.get(ctx.state, &validator)?
```

Comment:
nit: this will omit validators that have not missed any votes but the comments say "all consensus validators". Let's either update the comments or default this to 0

Reply:
agree, I would prefer to include all validators and default to 0

Reply:
fixed in [16e92a8](https://github.com/anoma/namada/pull/3899/commits/16e92a8e716feca13113d7956082fd41b65043a2)

Reply:
Actually I considered the value to default like 0, but it has a posibility to validators about misunderstand. the default 0 means for validator to haven't any missed. So I chose to exclude the value rather than the default value if this command(`sum_liveness_handle.get(ctx.state, &validator)?`) failed for some reason.

Reply:
if the result of this is `None` rather than `Some(missed_votes)` it means that there are no missed votes, not a failure in a command (a possible failure is handled via `?` at the end of the call)

Reply:
Ok, understood Thanks @tzemanovic